lib/repository/cat/cat_repository_impl.dart

```diff
@@ -1,22 +1,20 @@
+import 'package:flutter_template/repository/cat/cat_repository.dart';
 import 'package:flutter_template/repository/cat/domain_cat_mapper.dart';
-import 'package:flutter_template/services/cat/remote/cat_remote_service.dart';
 import 'package:flutter_template/services/entity/cat/remote/remote_cat.dart';
 
-class CatRepositoryImpl implements CatRemoteService {
-  late final CatRemoteService catRemoteService;
-
-  late final DomainCatMapper domainCatMapper;
+class CatRepositoryImpl implements CatRepository {
+  final CatRepository catRepository;
+  final DomainCatMapper domainCatMapper;
```

Comment:
I think we need to inject the remote service here.
Injecting the same class will cause a cyclic dependency.

Reply:
Checking on this front